### DISCUSSION TITLE
More flexible order of mobjects in the scene.

### DISCUSSION THREAD

**Discussion by Morkunas** [ManimCommunity/manim#2802] [Suggestions and Proposals]
## Description of proposed feature
It would be amazing, if we were able to specify more flexible order of mobjects in the scene. 
For example: if mob1 was behind mob2, mob2 was behind mob3 and mob3 was behind mob1.
Currently, one mobject will always be behind all others.

Here is a visual example. 
I think, the best we can do right now is something like this. Note how pink circle is in the back.
![FourCirclesOld](https://user-images.githubusercontent.com/20854503/139561657-e6c50875-dc1e-44be-8d60-676c618bd0db.png)

Ideally, it would be possible to achieve something like this. Note that each circle covers exactly one other circle:
![FourCircles_ManimCE_v0 6 0](https://user-images.githubusercontent.com/20854503/139561717-0dfedc69-4426-46e1-9cd3-9f717ab76e8f.png)

Thank you! 

## How can the new feature be used?
Mostly more flexibility in the scene where we have non trivial dependencies between mobjects.

**Comment by k4pran** [comment]
Are you using the cairo (default) renderer or opengl? There is a PR #2235 open currently to begin implementing something like this for opengl. We have also been having discussions on discord whether this idea could also work for cairo. I agree though that we should be able to have more control over order

**Comment by Morkunas** [comment]
Hey! Thanks for the response! Currently I am using Cairo! But this could be a good reason to move to opengl!
